Sources/Classes/RSEventRepository.m

```diff
@@ -146,7 +146,11 @@ - (void) __initiateSDK {
                     strongSelf->isSDKEnabled = serverConfig.isSourceEnabled;
                 });
                 if  (strongSelf->isSDKEnabled) {
-                    self->dataPlaneUrl = [RSUtils getDataPlaneUrlFrom:serverConfig andRSConfig:self->config];
+                    strongSelf->dataPlaneUrl = [RSUtils getDataPlaneUrlFrom:serverConfig andRSConfig:self->config];
+                    if (strongSelf->dataPlaneUrl == nil) {
+                        [RSLogger logError:@"Invalid dataPlaneUrl: The dataPlaneUrl is not provided or given dataPlaneUrl is not valid\n**Note: dataPlaneUrl is mandatory for Free Plan users from version 1.11.0**"];
+                        break;
```

Comment:
I believe using `return` instead of `break` would carry more significance as it indicates an immediate return.

Reply:
We can do this.